### PR TITLE
Update date format docs with BigQuery SQL

### PR DESCRIPTION
**Pull request by jklukas**

### CODE REVIEW

File: src/concepts/analysis_gotchas.md [hunk 1]
```diff
@@ -126,25 +126,26 @@ ask them to install Beta.
 
 ### Trusting Dates
 
-Don't trust client times.
+Don't trust desktop client times. The situation is somewhat better on mobile devices
+and being investigated.
```
Comment (chutten) [inline src/concepts/analysis_gotchas.md:130]:
```suggestion
and is being investigated.
```
If there's a bug for the investigation, here'd be a good place to link to it.

File: src/concepts/analysis_gotchas.md [hunk 2]
```diff
@@ -126,25 +126,26 @@ ask them to install Beta.
...
 The user's clock can be set (purposefully or accidentally) to any time at all.
 The nature of SSL certificates tends to keep this within a certain relatively-accurate window,
 because a user who's clock is too far in the past or too far in the future
 might confuse certain expiration-date-checking code.
 
-Examples of client times: `crashDate`, `crashTime`, `meta/Date`, `sessionStartDate`,
+Examples of client times from Firefox desktop pings: `crashDate`, `crashTime`, `meta/Date`, `sessionStartDate`,
 `subsessionStartDate`, `profile/creationDate`
 
+Examples of client times from Glean pings: `ping_info.end_time`
```
Comment (chutten) [inline src/concepts/analysis_gotchas.md:141]:
```suggestion
Examples of client times from Glean pings: [`ping_info.end_time`](https://mozilla.github.io/glean/book/user/pings/index.html#the-ping_info-section)
```

File: src/concepts/analysis_gotchas.md (hunk 1 above)
Comment (jklukas) [reply src/concepts/analysis_gotchas.md:130]: I knew you'd ask this. There is not a specific bug, but this is generally an important area for understanding day-by-day retention in Fenix that is getting a lot of interest right now.

File: src/concepts/analysis_gotchas.md [hunk 3]
```diff
@@ -168,10 +169,11 @@ Build ids look like dates but aren't.
 If you take the first eight characters you can use that as a proxy
 for the day the build was released.
 
-`metadata/Date` is an HTTP Date header in a [RFC 7231]-compatible format.
+`metadata.header.date` is an HTTP Date header in a [RFC 7231]-compatible format.
 
 **Tip:** To parse `metadata/Date` to a usable date in SQL:
-`DATE_PARSE(SUBSTR(client_submission_date, 1, 25), '%a, %d %b %Y %H:%i:%s')`
+`SAFE.PARSE_TIMESTAMP('%a, %d %b %Y %T %Z', REPLACE(metadata.header.date, 'GMT+00:00', 'GMT'))`
+or better yet use the pre-parsed version available in user-facing views `metatdata.header.parsed_date`
```
Comment (jklukas) [inline src/concepts/analysis_gotchas.md:176]:
```suggestion
or better yet use the already parsed version available in user-facing views (`metatdata.header.parsed_date`)
```

**Pull request merged**